pkg/apis/apps/validation/validation.go

```diff
@@ -75,6 +75,48 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path) fi
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("podManagementPolicy"), spec.PodManagementPolicy, fmt.Sprintf("must be '%s' or '%s'", apps.OrderedReadyPodManagement, apps.ParallelPodManagement)))
 	}
 
+	switch spec.UpdateStrategy.Type {
+	case "":
+		allErrs = append(allErrs, field.Required(fldPath.Child("updateStrategy"), ""))
+	case apps.OnDeleteStatefulSetStrategyType, apps.RollingUpdateStatefulSetStrategyType:
+		if spec.UpdateStrategy.Partition != nil {
+			allErrs = append(
+				allErrs,
+				field.Invalid(
+					fldPath.Child("updateStrategy").Child("partition"),
+					spec.UpdateStrategy.Partition.Ordinal,
+					fmt.Sprintf("only allowed for updateStrategy '%s'", apps.PartitionStatefulSetStrategyType)))
+		}
+	case apps.PartitionStatefulSetStrategyType:
+		if spec.UpdateStrategy.Partition == nil {
```

Comment:
Would we have staged rollout marked as completed when we move partition inside the RollingUpdate struct ? I would like to see the same method of determining whether a Update completed or not , whether we are using staged or not. I would canary for 2 days or even 1 week, and having to determine the update completed based on looking at different fields is confusing not to say that a external tooling/portal showing the staged rollout as incomplete until the partitioning is removed and normal rolling update strategy is applied.